text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1258 @@
+      type: AttributeType.STRING,
+    },
+    sortKey: {
+      name: 'FooRangeKey',
+      type: AttributeType.STRING,
+    },
+    billingMode: BillingMode.provisioned({
+      writeCapacity: Capacity.autoscaled({ max: 70 }),
+      readCapacity: Capacity.autoscaled({ max: 50 }),
+    }),
+    globalSecondaryIndex: [{
+      indexName: 'UniqueGsiName',
+      partitionKey: {
+        name: 'FooRangeKey',
+        type: AttributeType.STRING,
+      },
+      readCapacity: Capacity.autoscaled({ max: 50 }),
+    }],
+    replicas: [{
+      region: 'us-east-1',
+      globalSecondaryIndexOptions: [
+        {
+          indexName: 'UniqueGsiName',
+          readCapacity: Capacity.fixed(55),
+        },
+      ],
+    }],
+  });
+  ```
+
+__NOTE:__
+
+* You can create up to 20 global secondary indexes.
+* You can only create or delete one global secondary index in a single stack operation.
+
+#### Local Secondary Indexes
+
+You can add local secondary indexes to your global table. These will be the same for each replica of
+the table.
+
+```typescript
+const globalTable = new GlobalTable(tableStack, 'FooTable', {
+  tableName: 'FooTable',
+  partitionKey: { name: 'Foo', type: AttributeType.STRING },
+  sortKey: { name: 'Bar', type: AttributeType.STRING },
+  localSecondaryIndex: [
+    {
+      indexName: 'FooTableLsi',
+      sortKey: { name: 'Foo', type: AttributeType.STRING },
+    },
+  ],
+});
+```
+
+You can add local secondary index with `addLocalSecondaryIndex` method.
+
+```typescript
+globalTable.addLocalSecondaryIndex({
+  indexName: 'FooTableLsi',
+  sortKey: { name: 'Foo', type: AttributeType.STRING },
+});
+```
+
+__NOTE:__
+
+* You need a sort key to define a local secondary index.
+* You can create up to five local secondary indexes.
+
+#### Encryption
+
+Encryption mode defines how the table, replicas and GSIs would be encrypted at rest. There are now four
+types of encryptions that are available for global tables:
+
+1. `AWS_OWNED`: This uses a KMS key for encryption that is owned by DynamoDB. This is the default
+for global tables.
+2. `AWS_MANAGED`: A KMS key is created in your account and is managed by AWS.
+3. `KEY_ARNS`: [NEW] You will provide KMS key arns for each replica. You will need to specify arns for each
+replica region. The key also needs to be in the same region as the replica.
+4. `MULTI_REGION_KEY`: [NEW] A multi region KMS key and its supporting stacks in replica regions will be
+provisioned automatically.
+
+The `encryption` mode selected remains the same for each replica. If you will like to provide KMS keys managed
+by you for each replica, then you can use the `KEY_ARNS` encryption mode and provide region specific key in
+`encryptionKey` property of each replica.
+
+```typescript
+new GlobalTable(tableStack, 'FooTable', {
+  tableName: 'FooGlobalTable',
+  partitionKey: {
+    name: 'FooHashKey',
+    type: AttributeType.STRING,
+  },
+  encryption: GlobalTableEncryption.KEY_ARNS,
```

Comment:
Don't forget the keyARN for the table itself. Which probably shouldn't be an ARN but an `IKey`.